### PR TITLE
Discovery service

### PR DESCRIPTION
**Pull request by dylanlott**
Had a chat with Egon about structuring the Overlay service differently. 

Egon pointed out that the Overlay doesn't need to talk to Kad, it only needs to listen for what Kad tells it, meaning we could separate out some of the refresh and bootstrapping logic into it's own service (pkg/discovery) and that would make it easier for other services to tie into node discovery without requiring an event system or callbacks.

The main motivation behind this idea was cyclical import errors I was starting to get, which is usually a code smell for a structure problem.

### CODE REVIEW

File: pkg/discovery/server.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+package discovery
+
+import (
+	"go.uber.org/zap"
+	"gopkg.in/spacemonkeygo/monkit.v2"
+)
+
+// Server struct that loads on logging and metrics
+type Server struct {
+	logger  *zap.Logger
```
Comment (egonelbre) [inline pkg/discovery/server.go]: Let's call this `log`

File: pkg/overlay/config.go [hunk 2]
```diff
@@ -92,29 +92,29 @@ func (c Config) Run(ctx context.Context, server *provider.Provider) (
...
-				err := cache.Refresh(ctx)
-				if err != nil {
-					zap.L().Error("Error with cache refresh: ", zap.Error(err))
-				}
-			case <-ctx.Done():
-				return
-			}
-		}
-	}()
+	//go func() {
```
Comment (egonelbre) [inline pkg/overlay/config.go]: Can we remove these comments?

File: pkg/discovery/server.go (hunk 1 above)
Comment (dylanlott) [reply pkg/discovery/server.go]: Done

File: pkg/overlay/config.go (hunk 2 above)
Comment (dylanlott) [reply pkg/overlay/config.go]: Removed.

**Pull request merged**